ring/ring.go

```diff
@@ -564,12 +571,46 @@ func (r *Ring) countTokens() (map[string]uint32, map[string]uint32) {
 	return numTokens, owned
 }
 
-// updateRingMetrics updates ring metrics.
+// countTokens returns the number of tokens and tokens within the range for each instance.
+func countTokensUtil(ringDesc *Desc) (map[string]uint32, map[string]uint32, int) {
+	ringTokens := ringDesc.GetTokens()
+	ringInstanceByToken := ringDesc.getTokensInfo()
+	owned := map[string]uint32{}
+	numTokens := map[string]uint32{}
+	for i, token := range ringTokens {
+		var diff uint32
+
+		// Compute how many tokens are within the range.
+		if i+1 == len(ringTokens) {
+			diff = (math.MaxUint32 - token) + ringTokens[0]
```

Comment:
is it guaranteed that the total ring size is always `math.MaxUint32`?

Reply:
Looks like it, because there are various places in the code which use `math.MaxUint32` in a similar way.

Reply:
I guess this is more of a convention that for example lifecycler and basic lifecycler follows. Not sure why you'd ever reduce the domain though as that just increases the chance of hash collisions.

Reply:
> is it guaranteed that the total ring size is always math.MaxUint32?

`math.MaxUint32` is the tokens space, which is fixed.